scripts/generate-components-handlers/event-handlers-parser.ts

```diff
@@ -0,0 +1,205 @@
+    return interfaceNode.members
+      .filter((member): member is ts.MethodSignature =>
+        isMemberEventHandler(member as ts.MethodSignature)
+      )
+      .map(member => buildEventHandler(rootInterfaceName, member));
+  };
+
+  const getBaseClassesEventHandlers = (
+    rootInterfaceName: string,
+    interfaceNode: ts.InterfaceDeclaration
+  ): EventHandler[] => {
+    const baseClassesNodes = getBaseClassesNames(interfaceNode)
+      .map(baseClassName => getStatementByInterfaceName(baseClassName))
+      .filter((statement): statement is ts.InterfaceDeclaration =>
+        Boolean(statement)
+      );
+
+    return baseClassesNodes
+      .map(baseClassNode => getEventHandlers(rootInterfaceName, baseClassNode))
+      .reduce<EventHandler[]>(mergeEventHandlers, []);
+  };
+
+  const getEventHandlers = (
+    rootInterfaceName: string,
+    interfaceNode: ts.InterfaceDeclaration
+  ): EventHandler[] => {
+    const interfaceName = interfaceNode.name.getText();
+    if (completed[interfaceName]) {
+      return handlersCache[interfaceName];
+    }
+
+    const rootMembers = getRootEventHandlers(rootInterfaceName, interfaceNode);
+    const baseHandlers = getBaseClassesEventHandlers(
+      rootInterfaceName,
+      interfaceNode
+    );
+
+    const componentHandlers = mergeEventHandlers(
+      rootMembers,
+      baseHandlers
+    ).map(e => overrideOriginWithRootInterface(e, rootInterfaceName));
+
+    handlersCache[interfaceName] = componentHandlers;
+    completed[interfaceName] = true;
+    return componentHandlers;
+  };
+
+  const getEventHandlersModuleBody = (
+    sourceFile: ts.SourceFile
+  ): ts.ModuleBlock => {
+    const eventHandlersModule = sourceFile.statements.find(
+      (statement): statement is ts.ModuleDeclaration => {
+        return (
+          ts.isModuleDeclaration(statement) &&
+          statement.name.text === Constants.EVENTS_INTERFACE_NAME
+        );
+      }
+    );
+
+    if (
+      !eventHandlersModule ||
+      !eventHandlersModule?.body ||
+      !ts.isModuleBlock(eventHandlersModule?.body)
+    ) {
+      throw new Error(
+        `Failed finding the ${Constants.EVENTS_INTERFACE_NAME} interface`
+      );
+    }
+    return eventHandlersModule.body;
+  };
+
+  const generateComponentsMapEventHandlers = (): ComponentsEventHandlers => {
+    eventHandlersModuleBody = getEventHandlersModuleBody(sourceFile);
+    const interfaces = eventHandlersModuleBody.statements.filter(
+      (statement): statement is ts.InterfaceDeclaration => {
+        return ts.isInterfaceDeclaration(statement);
+      }
+    );
+
+    return interfaces.reduce((componentsMap, interfaceNode) => {
+      const interfaceNodeName = interfaceNode.name.getText();
+      return Object.assign(componentsMap, {
+        [interfaceNodeName]: getEventHandlers(interfaceNodeName, interfaceNode)
+      });
+    }, {});
+  };
+
+  return {
+    parse: generateComponentsMapEventHandlers
+  };
+};
+
+export default getEventHandlersParser;
```

Comment:
i'm not able to get deep enough into this code... please show it to someone from the team to make sure it's understandable.